### ISSUE TITLE
Several endpoints are not using the user-friendly assets params

### ISSUE THREAD

**Issue by miguelnietoa** [kommitters/stellar_sdk#225]
### Precheck

- [x] Do a quick search and make sure the bug has not been reported yet.
- [x] Please disclose security vulnerabilities privately at [oss@kommit.co](mailto:oss@kommit.co).
- [x] Finally, be nice and have fun!

### Description
Several endpoints are not using the user-friendly assets params.

### Environment

* Elixir & Erlang/OTP versions (elixir --version): `Elixir 1.13.4 (compiled with Erlang/OTP 24)`
* Operating system: `Linux Ubuntu 22.04 LTS`

### Current behavior

```elixir
Stellar.Horizon.Offers.all(
  selling_asset_type: "credit_alphanum4",
  selling_asset_issuer: "GC2MTRMAXEIXRNY4ZCVNJ4DGBUQA2KTSXWDOQIIDGZZ3CKY5OYVJ4O56",
  selling_asset_code: "MTK"
)
```
### Expected behavior
The input fields with `_asset_type`, `_asset_issuer`, `_asset_code` should be refactored in the next way:

```elixir
Stellar.Horizon.Offers.all(
  selling_asset: [
    code: "MTK",
    issuer: "GC2MTRMAXEIXRNY4ZCVNJ4DGBUQA2KTSXWDOQIIDGZZ3CKY5OYVJ4O56"
  ]
)
```

### Additional context

- It should be implemented by using `RequestParams.build_assets_params/2`  to transform query options to be sent to Horizon API.
- It happens with several endpoints not only with `Offers`. It needs to be changed in every endpoint and asset input where this applies.

**Comment by miguelnietoa** [comment]
`Offers` and `Trades` have implemented this user-friendly way in [v0.10.2](https://github.com/kommitters/stellar_sdk/releases/tag/v0.10.2) 😄 . 

Thanks @SanGeeky

**Comment by juanhurtado10** [comment]
@SanGeeky Thanks for contributing 🚀 !!
